crates/sui-indexer/src/lib.rs

```diff
@@ -201,43 +198,24 @@ impl Indexer {
                 .get_latest_tx_checkpoint_sequence_number()
                 .await
                 .expect("Failed to get latest tx checkpoint sequence number from DB");
-            let (downloaded_checkpoint_data_sender, downloaded_checkpoint_data_receiver) =
-                mysten_metrics::metered_channel::channel(
-                    DOWNLOAD_QUEUE_SIZE,
-                    &mysten_metrics::get_metrics()
-                        .unwrap()
-                        .channels
-                        .with_label_values(&["checkpoint_tx_downloading"]),
-                );
-
-            // experimental rest api route is found at `/rest` on the same interface as the jsonrpc
-            // service
-            let rest_api_url = format!("{}/rest", config.rpc_client_url);
-            let fetcher = CheckpointFetcher::new(
-                sui_rest_api::Client::new(&rest_api_url),
-                if last_seq_from_db < 0 {
-                    None
-                } else {
-                    Some(last_seq_from_db as u64)
-                },
-                downloaded_checkpoint_data_sender,
-            );
-            spawn_monitored_task!(fetcher.run());
+            let last_downloaded_checkpoint = if last_seq_from_db < 0 {
+                None
+            } else {
+                Some(last_seq_from_db as u64)
+            };
 
             let (checkpoint_handler, object_handler) = new_handlers(store, metrics, config);
 
-            crate::framework::runner::run(
-                mysten_metrics::metered_channel::ReceiverStream::new(
-                    downloaded_checkpoint_data_receiver,
-                ),
-                vec![Box::new(checkpoint_handler), Box::new(object_handler)],
-            )
-            .await;
-
-            Ok(())
-        } else {
-            Ok(())
+            IndexerBuilder::new()
```

Comment:
so this is how a generic indexer would start, right?
I mean a binary with that code....

Reply:
yes correct, you'd just create your own `main.rs` and initialize this IndexerBuilder and go to town